src/main/java/com/uber/cadence/internal/sync/ActivityInvocationHandler.java

```diff
@@ -43,9 +44,30 @@ private ActivityInvocationHandler(ActivityOptions options, WorkflowInterceptor a
   @Override
   protected Function<Object[], Object> getActivityFunc(
       Method method, MethodRetry methodRetry, ActivityMethod activityMethod, String activityName) {
+
+    ActivityOptions activityOptionsOverride = null;
+    WorkflowThread workflowThread = WorkflowInternal.getRootWorkflowContext();
+    if (workflowThread.getDecisionContext() != null
+        && workflowThread.getDecisionContext().getWorkflowImplementationOptions() != null) {
+      Map<String, ActivityOptions> activityOptionsMap =
+          workflowThread
+              .getDecisionContext()
+              .getWorkflowImplementationOptions()
+              .getActivityOptions();
+
+      if (activityOptionsMap.containsKey(activityName)) {
+        activityOptionsOverride = activityOptionsMap.get(activityName);
+      }
+    }
+
     Function<Object[], Object> function;
     ActivityOptions mergedOptions = ActivityOptions.merge(activityMethod, methodRetry, options);
-    ActivityStub stub = ActivityStubImpl.newInstance(mergedOptions, activityExecutor);
+    ActivityStub stub;
+    if (activityOptionsOverride == null) {
+      stub = ActivityStubImpl.newInstance(mergedOptions, activityExecutor);
+    } else {
+      stub = ActivityStubImpl.newInstance(activityOptionsOverride, activityExecutor);
```

Comment:
Might makes sense to merge override options as well.

Reply:
yeah, as discussed, only if there's a sense in which the interface options are 'default', if I've understood that correctly. It might be more ergonomic to just override specific options rather than the entire set.